crates/type-system/Cargo.toml

```diff
@@ -0,0 +1,40 @@
+[package]
+name = "type-system"
+version = "0.0.0"
+edition = "2021"
+authors = ["HASH"]
+publish = false
+description = "Definitions of types within the Block Protocol Type System"
+
+[lib]
+name = "type_system"
+crate-type = ["cdylib", "rlib"]
+
+[features]
+default = ["console_error_panic_hook"]
+
+[dependencies]
+serde = { version = "1.0.137", features = ["derive"] }
+thiserror = "1.0.32"
+url = "2.2.2"
+wasm-bindgen = { version = "0.2.63", features = ["serde-serialize"] }
+
+# The `console_error_panic_hook` crate provides better debugging of panics by
+# logging them with `console.error`. This is great for development, but requires
+# all the `std::fmt` and `std::panicking` infrastructure, so isn't great for
+# code size when deploying.
+console_error_panic_hook = { version = "0.1.6", optional = true }
+
+# `wee_alloc` is a tiny allocator for wasm that is only ~1K in code size
+# compared to the default allocator's ~10K. It is slower than the default
+# allocator, however.
+#
+# Unfortunately, `wee_alloc` requires nightly Rust when targeting wasm for now.
+wee_alloc = { version = "0.4.5", optional = true }
```

Comment:
Given the size of the binary is it worth pulling an external allocator for this?

Reply:
I think so, yes. We're very likely to address the binary size problems from `url` and generally speaking a lot of people seem to use `wee_alloc` for WASM so I'm happy to stick with convention.